libs/clients/rsk/personal-tax-return/src/lib/requests/pdf.ts

```diff
@@ -0,0 +1,21 @@
+export const pdfRequest = (
+  agentNationalId: string,
+  agentId: string,
+  requesterNationalId: string,
+  year: string,
+) => {
+  return `<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:tem="http://tempuri.org/" xmlns:ns="http://skattur.is/UmbodsmadurSkuldaraThjonusta/2011/01/03">
+  <soap:Header xmlns:wsa="http://www.w3.org/2005/08/addressing"><wsa:Action>http://tempuri.org/IUSStadgreidslaFramtalGogn/SaekjaPDFAfritFramtalsEinstaklings</wsa:Action><wsa:To>https://vefurp.rsk.is/ws/securep/UMS/WS/USStadgreidslaFramtalGogn.svc</wsa:To></soap:Header>
+  <soap:Body>
+     <tem:SaekjaPDFAfritFramtalsEinstaklings>
+        <!--Optional:-->
+        <tem:inntak>
+           <ns:KennitalaUmbodsmanns>${agentNationalId}</ns:KennitalaUmbodsmanns>
+           <ns:AudkenniUmbodsmanns>${agentId}</ns:AudkenniUmbodsmanns>
+           <ns:KennitalaSkuldara>${requesterNationalId}</ns:KennitalaSkuldara>
+           <ns:Tekjuar>${year}</ns:Tekjuar>
```

Comment:
Should defend against XML injection. I much prefer setting up some kind of WSDL-to-TypeScript workflow to make soap integrations nicer.

No-one has done this before in the monorepo unfortunately, but we did set up something like that in the vidspyrna repo, so it's definitely doable.

Reply:
Implemented in a naive manner, would love feedback on it @eirikurn and WSDL-to-TS sounds nice, never seen, do you have an example which I could look at for curiosity?

Reply:
My concern was mostly to escape the input, eg using `_.escape()`. Sanitization gets the job done though. Thankfully there is also no user input in this case, but always good to escape. Data validation can occur in other places (ideally as late as possible).

As I said, there is an example in [Vidspyrna here](https://github.com/island-is/vidspyrna/blob/de193ce0dac722a818596da595d20dabd9ab247d/apps/api/src/services/rsk/soap/types.ts). Haven't looked more into it though.